### ISSUE TITLE
Start wave

### ISSUE THREAD

**Issue by alketii** [KOBUGE-Games/project-kumquat#19]
The "Next wave" button should initially be "Start", so you will have time to plan a strategy and place towers.

**Comment by ObaniGemini** [comment]
I don't think so, as you can launch next wave while the other isn't totally over.

**Comment by akien-mga** [comment]
> The "Next wave" button should initially be "Start", so you will have time to plan a strategy and place towers.

That's worth discussing, at it brings the question of whether we are going more for a arcade/time-constrained gameplay or more tactical.
- Arcade/time-constrained: starting the game would start the timer for the first wave, and the waves would come by themselves one after the other, so you need to keep the pace and ensure that you are always able to fight off the coming monsters. You can place "Next wave" if you're feeling overpowered and want to bring the next wave earlier than the normal timer would.
- Tactical: you can place your towers before starting, and start the first wave by clicking the button. Then you need to click the button again if you want to start the following wave, i.e. you have unlimited time to plan your towers and upgrades for the next wave.

I haven't played many TDs, but those I played were more of the arcade style; it's likely easier for balancing as you just need to make it very hard, and then balance the game so that it's beatable if you're good :D With tactical we would likely have to make levels that are more puzzle-like and require that you think a bit between each wave about how best to choose and place your towers (it could e.g. give hints about incoming monsters of the next wave so that you know which towers are best suited).

Both types have their merit I think, tactical is maybe more interesting but arcade is simpler :D Or we could go for a kind of mix maybe. Opinions?

**Comment by bojidar-bg** [comment]
I'm all for tactical, but, because everybody has a different opinion, and consensus would be hard, what about putting up a "Settings" page for it (along with other stuff)? :smile:

**Comment by akien-mga** [comment]
> I'm all for tactical, but, because everybody has a different opinion, and consensus would be hard, what about putting up a "Settings" page for it (along with other stuff)? :smile:

IMO it would have a big impact on balancing and how we design the levels, so a setting would probably not be satisfying, the game would always be unbalanced in one mode or the other. On the other hand we could code the two modes, and propose different game rules for each mode (including different levels and different creature and tower balancing); one would choose the mode directly in the main menu and not in the settings page (i.e. Play > Arcade and Play > Tactical).

That could be nice :) We should however focus on one mode for now IMO, to try to get one playable mode before we work on rebalancing everything for the other one :)

**Comment by ObaniGemini** [comment]
I think we should start with an arcade mode, that would not be so much considering strategy, but more adaptation skills. As @akien-mga said, going for tactical would end with us trying to make puzzle level, and optimizing ennemies number and map organization.No

We shall first work on the game in its content, and to work on it, I think it's better to have an arcade-mode game. If some of us has time and faith to do some puzzle level(s), it's ok. But I think it would be more boring than anything else and that it would take too much time :p

I agree with the idea of @bojidar-bg of making a settings menu, but again, I think this should come later ^^ (logically).

`We should however focus on one mode for now IMO, to try to get one playable mode before we work on rebalancing everything for the other one :)`

**Yes**

**Comment by fry-** [comment]
I vote for **tactical** because of several reasons.
1. All TD games I know of use the arcade style. It's time for something new :smile: 
2. We could include the "twist" that way. Before you click "next wave", you have to first choose the towers that get disabled because of monsters that entered the nexus.
3. I will add beacons to the enemy-spawning spots. Now, before next wave starts, you could see which portals are open (where enemies spawn for next wave) and you can arrange your towers accordingly. with arcade style, this would be too stressful and thus not really possible.

PS: @ObaniGemini has a point, but I think we can do good levels and balance it well. I have faith in us :smile_cat:

**Comment by akien-mga** [comment]
Alright, I agree to go with tactical then and try to do something a bit original instead of a boring typical arcade TD :D

**Comment by ObaniGemini** [comment]
Ok, so let's do this :)
`All TD games I know of use the arcade style. It's time for something new :smile:`
Lol for me it's the whole opposite ^^
